samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/ExternalInformationSkill.cs

```diff
@@ -122,24 +122,9 @@ public async Task<string> AcquireExternalInformationAsync(string userIntent, SKC
 
             if (plan.Steps.Count > 0)
             {
-                // Merge any variables from the context into plan's state
-                // as these will be used on plan execution.
-                // These context variables come from user input, so they are prioritized.
-                var variables = context.Variables;
-                foreach (var param in plan.State)
-                {
-                    if (param.Key.Equals("INPUT", StringComparison.OrdinalIgnoreCase))
-                    {
-                        continue;
-                    }
-
-                    if (variables.TryGetValue(param.Key, out string? value))
-                    {
-                        plan.State.Set(param.Key, value);
-                    }
-                }
-
-                this.ProposedPlan = plan;
+                // Parameters stored in plan's top level state
+                this.MergeContextIntoPlan(context.Variables, plan.State);
+                this.ProposedPlan = this.SanitizePlan(plan, context);
```

Comment:
Let's add a TODO that the kernel should improve to give developer options here and have this behavior be the default.

Reply:
Done